packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts

```diff
@@ -0,0 +1,181 @@
+import { GraphQLClient } from 'graphql-request';
+import {
+  DEFAULT_PROTECTED_DATA_SHARING_APP,
+  DEFAULT_SHARING_CONTRACT_ADDRESS,
+} from '../../config/config.js';
+import { toHex } from '../../utils/data.js';
+import { ErrorWithData, WorkflowError } from '../../utils/errors.js';
+import {
+  addressOrEnsOrAnySchema,
+  positiveNumberSchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
+import {
+  Address,
+  BuyProtectedDataParams,
+  IExecConsumer,
+  SharingContractConsumer,
+  SubgraphConsumer,
+  SuccessWithTransactionHash,
+} from '../types/index.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+import { getCollectionById } from './subgraph/getCollectionById.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
+
+export async function buyProtectedData({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  sharingContractAddress = throwIfMissing(),
+  protectedDataAddress,
+  collectionTokenIdTo,
+  appAddress,
+}: IExecConsumer &
+  SubgraphConsumer &
+  SharingContractConsumer &
+  BuyProtectedDataParams): Promise<SuccessWithTransactionHash> {
+  try {
+    const vProtectedDataAddress = addressOrEnsOrAnySchema()
+      .required()
+      .label('protectedDataAddress')
+      .validateSync(protectedDataAddress);
+
+    const vCollectionTokenIdTo = positiveNumberSchema()
+      .label('collectionTokenIdTo')
+      .validateSync(collectionTokenIdTo);
+
+    const vAppAddress = addressOrEnsOrAnySchema()
+      .label('appAddress')
+      .validateSync(appAddress);
+
+    const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+
+    const protectedData = await checkAndGetProtectedData({
+      graphQLClient,
+      protectedDataAddress: vProtectedDataAddress,
+      userAddress,
+    });
+
+    let collection;
+    if (vCollectionTokenIdTo) {
+      collection = await checkAndGetCollection({
+        graphQLClient,
+        collectionTokenId: vCollectionTokenIdTo,
+        userAddress,
+      });
+    }
+
+    const sharingContract = await getSharingContract(
+      iexec,
+      sharingContractAddress
+    );
+
+    let tx;
+    if (collection) {
+      tx = await sharingContract.buyProtectedDataForCollection(
+        protectedData.collection.id, // _collectionTokenIdFrom
+        protectedData.id,
+        collection.id, // _collectionTokenIdTo
+        vAppAddress || DEFAULT_PROTECTED_DATA_SHARING_APP,
+        {
+          // TODO: See how we can remove this
+          gasLimit: 900_000,
+        }
```

Comment:
you missed the value field : 
```suggestion
        {
           value: // price of the sale
          // TODO: See how we can remove this
          gasLimit: 900_000,
        }
```

Reply:
Good catch indeed, thanks